Add PDF export option to reports (other than query reports)

martinzoller opened the pull request:
Inserted code adapted from query_report.js into report_view.js to allow for creating a PDF document from any report, including user-defined reports. Also added the option of using a placeholder ("{today}") in report-specific letter heads (header and footer part), which will be replaced by today's date when a report is exported.

Commit:
Add PDF export option to generic report view

Commit:
Add support for a printing date placeholder to the general printing template

lasalesi commented:
Tested the PR. Test case: Customer > Report Generator

Menu entry appears as expected:
![grafik](https://user-images.githubusercontent.com/26505250/171023388-e07711e4-ee7b-4588-bc3f-9a6552ac96db.png)

However, a click to Menu > PDF leads to no action. The browser console reports:

    Uncaught TypeError: e.report_doc is undefined

from report_view.js line 1198.

Commit:
Fix PDF export for unsaved reports; fix letter head setting

martinzoller commented:
I was able to reproduce the bug reported by @lasalesi:
> ```
> Uncaught TypeError: e.report_doc is undefined
> ```
I had not considered that report_doc is only defined once the report has been saved. This is now fixed.
In addition I fixed the default behavior of the "letter head" checkmark in the print settings: If the report document has a letter head assigned, the checkmark will be selected by default.
If no specific letter head is assigned, the checkmark remains unchecked, even if a company default letter head is defined.

lasalesi commented:
Retested and it works as expected. One comment: if there is no data, nothing happens and the console reports "datatable not defined"; this is acceptable, as there is no data > hence the export to pdf as no direct use.
